### PR TITLE
feat(sdk): generate sample react component with the embedding cli

### LINKED ISSUE #46417: Generate sample React component files via the embedding sdk's cli
Modify the embedding sdk's cli to generate React component files with cards that open the x-ray as embedded dashboards. 

- Provide an ability to switch between predefined themes
- Include a link to ”edit this dashboard” that can be deleted later

[Product doc.](https://www.notion.so/metabase/CLI-connects-DB-creates-models-and-X-rays-to-embed-07c8cdd62ebf4ec7989460e000f0b9c1?pvs=4)

### PR DESCRIPTION
**Pull request by heypoom**
Closes https://github.com/metabase/metabase/issues/46417

### Description

Generates example React components that demonstrates how to setup the embedding SDK.

### How to verify

- Run `yarn build-embedding-sdk-cli:watch` in one terminal, in your Metabase directory
- Create an empty React project, using `yarn create vite`
- Run `yarn add file:../metabase/resources/embedding-sdk` to add the dev version of the SDK
- Run `npx @metabase/embedding-sdk-react start`
- Go through the CLI's steps until the end. Watch the below demo video. Make sure to add a license key, either at the beginning or after the setup step.
- You should get the same result as in the demo video below.

### Demo

<div>
    <a href="https://www.loom.com/share/40edc8d096fe49e5ba1e6ea83631d9ff">
      <p>Embedding CLI - From Zero to Dashboard - Watch Video</p>
    </a>
    <a href="https://www.loom.com/share/40edc8d096fe49e5ba1e6ea83631d9ff">
      
    </a>
  </div>

### CODE REVIEW

File: enterprise/frontend/src/embedding-sdk/cli/constants/messages.ts [hunk 1]
```diff
@@ -27,13 +27,19 @@ export const EMBEDDING_FAILED_MESSAGE = `
...
-// eslint-disable-next-line no-unconditional-metabase-links-render -- link for the CLI message
-export const METABASE_INSTANCE_SETUP_COMPLETE_MESSAGE = `
+export const PREMIUM_TOKEN_REQUIRED_MESSAGE =
+  "  Don't forget to add your premium token to your Metabase instance in the admin settings! The embedding demo will not work without a license.";
+
+export const getMetabaseInstanceSetupCompleteMessage = (instanceUrl: string) =>
+  // eslint-disable-next-line no-unconditional-metabase-links-render -- link for the CLI message
+  `
   Metabase instance is ready for embedding.
+  Go to ${instanceUrl} to start using Metabase.
```
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/cli/constants/messages.ts:37]: I find myself re-typing the instance url often, so it's nice to show them at the end of the setup.

File: enterprise/frontend/src/embedding-sdk/cli/constants/messages.ts [hunk 2]
```diff
@@ -27,13 +27,19 @@ export const EMBEDDING_FAILED_MESSAGE = `
...
+
   You can find your login credentials at METABASE_LOGIN.json
   Don't forget to put this file in your .gitignore.
 
-  Metabase will phone home some data collected via Google Analytics and Snowplow.
```
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/cli/constants/messages.ts:36]: @npretto mentioned we do not use Google Analytics anymore, only Snowplow.

File: enterprise/frontend/src/embedding-sdk/cli/steps/setup-metabase-instance.ts [hunk 3]
```diff
@@ -94,6 +94,7 @@ export const setupMetabaseInstance: CliStepMethod = async state => {
 
     if (!res.ok) {
       const errorMessage = await res.text();
+      spinner.fail();
```
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/cli/steps/setup-metabase-instance.ts:97]: Without this, the spinner gets stuck on a loading state even when the request failed.

**Comment by heypoom** [comment]
(resolving merge conflicts)

**Review by deniskaber — APPROVED** [review]

File: enterprise/frontend/src/embedding-sdk/cli/steps/generate-component-files.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+  // Generate sample components files in the specified directory.
+  for (const { name, content } of sampleComponents) {
+    await fs.writeFile(`${path}/${name}.jsx`, content);
+  }
+
+  // Generate analytics.css sample styles.
+  await fs.writeFile(`${path}/analytics.css`, ANALYTICS_CSS_SNIPPET);
+
+  // Generate index.ts file with all the component exports.
```
[the remark this answers is not in the document]
Comment (heypoom) [reply enterprise/frontend/src/embedding-sdk/cli/steps/generate-component-files.ts:54]: Great catch, thanks!

File: enterprise/frontend/src/embedding-sdk/README.md [hunk 5]
```diff
@@ -45,6 +45,17 @@ Features not yet supported:
...
 
+## Quickstart
+
+Run `npx @metabase/embedding-sdk-react start` in your React project to get started.
+
+This will start a local Metabase instance with Docker and generate sample dashboard components from your data. First run of this command will take 1 - 2 minutes to install the dependencies.
+
+Prerequisites:
+
+- [Node.js 20.x LTS](https://nodejs.org/en) or higher
```
[the remark this answers is not in the document]
Comment (heypoom) [reply enterprise/frontend/src/embedding-sdk/README.md:56]: @npretto sounds good, I've been running the CLI fine on Node.js 20 though, so I'll make it `Node.js 18.x LTS or higher` then :+1:

**Pull request merged**